util/object_pool.c

```diff
@@ -260,6 +274,12 @@ int comdb2_objpool_resume(comdb2_objpool_t op)
     return 0;
 }
 
+int comdb2_objpool_clear(comdb2_objpool_t op)
+{
+    objpool_evict_all_int(op);
```

Comment:
You need to hold evict_mutex here, and then call op->evict_impl(). You don't need to write your own objpool_evict_all_int(). They're already implemented (different pool types have different eviction routines).

Reply:
Perhaps the objpool_evict_all_int() function is misnamed.  Really, it should be called something like objpool_clear()...  The intent is that it always completely empties out the pool of all objects that are NOT in use.  It seems the evict_impl() functions (potentially) have some additional criteria?